### PR TITLE
chore: Merge latest changes in master into auto-scaling per shard dev branch

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-289710

For verifying that the merging of changes coming from master was correct I verified the following:

- From this PR looked at all changes related to advanced_cluster, ensuring that changes coming from master did not impact any logic related to new CRUD logic for autoscaling.
- Looked at changes in https://github.com/mongodb/terraform-provider-mongodbatlas/compare/master...merge-lastest-master-changes comparing this branch with master and ensuring no functionality from master was deleted and only autoscaling changes are present.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 1]
```diff
@@ -1153,8 +1153,8 @@ func checkIndependentShardScalingMode(clusterName, expectedMode string) resource
 		if rs.Primary.ID == "" {
 			return fmt.Errorf("no ID is set")
 		}
-		ids := conversion.DecodeStateID(rs.Primary.ID)
-		issMode, _, err := acc.GetIndependentShardScalingMode(context.Background(), ids["project_id"], clusterName)
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:1004]: This change was needed as `advancedclustertpf` implementation does not populate `id` value for the resource. @lantoli @EspenAlbert should we consider populating the with the same encoded value as in SDK v2 implementation? Don't see a use case for using the encoded value, but just to be on the conservative side.

**Comment by AgustinBettati** [comment]
ignoring `ldap` and `stream` flaky failing test groups as dev branch has no changes in these resources. 
For advanced_cluster ran `TestAccClusterAdvancedCluster_advancedConfig_oldMongoDBVersion` locally.
`2024-12-13T12:17:29.374+0100 [DEBUG] sdk.helper_resource: Finished TestCase: test_name=TestAccClusterAdvancedCluster_advancedConfig_oldMongoDBVersion
PASS`

File: internal/service/advancedcluster/resource_advanced_cluster_test.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/service/advancedcluster/resource_advanced_cluster_test.go:1004]: IMO: We should not populate the `id` this was always meant as an internal field

**Review by EspenAlbert — APPROVED** [review]
LGTM

**Review by oarbusi — APPROVED** [review]

**Pull request merged**